backend/authentication-microservice/src/main/java/com/ss/utopia/auth/service/UserService.java

```diff
@@ -71,70 +72,44 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
    */
   public User signup(UserDto userDto) {
     LOGGER.info("New user attempting to sign up");
-    User user = null;
-    Long roleId = 0L;
-    if (userDto.getRole().toUpperCase().equals("ADMIN")) {
-      roleId = 1L;
-    } else if (userDto.getRole().toUpperCase().equals("CUSTOMER")) {
-      roleId = 2L;
-    } else if (userDto.getRole().toUpperCase().equals("AGENT")) {
-      roleId = 3L;
-    } else {
-      throw new HttpServerErrorException(HttpStatus.BAD_REQUEST, "Unauthorized Role");
-    }
-    User createdUser = new User(userDto.getUsername(), passwordEncoder.encode(userDto.getPassword()), new UserRole(roleId, userDto.getRole()),
-      userDto.getGivenName(), userDto.getFamilyName(), userDto.getEmail(), userDto.getPhone());
-
-    try {
-      user = userDao.save(createdUser);
-    } catch (Exception e) {
-      String error = e.getMessage();
-      if (error.contains("user.email_UNIQUE")) {
-        throw new HttpServerErrorException(HttpStatus.BAD_REQUEST, "Email is taken");
-      } else if (error.contains("user.username_UNIQUE")) {
-        throw new HttpServerErrorException(HttpStatus.BAD_REQUEST, "Username is taken");
-      } else if (error.contains("user.phone_UNIQUE")) {
-        throw new HttpServerErrorException(HttpStatus.BAD_REQUEST, "Phone number is taken");
-      }
-    }
-    return user;
+    userDto.setPassword(passwordEncoder.encode(userDto.getPassword()));
+    User user = new User(userDto);
+    
+    if(user.getRole().equals(null))
```

Comment:
This will never be null but getRole().getId() will be null

Reply:
Yea, I noticed that and pushed right before you commented this, lol